elyra/pipeline/component_registry.py

```diff
@@ -133,61 +128,41 @@ def get_component(self, processor_type, component_id):
+                                        name="Notebook",
+                                        description="Notebook file",
+                                        op="execute-notebook-node"),
+                              Component(id="python-script",
+                                        name="Python Script",
+                                        description="Python Script",
+                                        op="execute-python-node"),
+                              Component(id="r-script",
+                                        name="R Script",
+                                        description="R Script",
+                                        op="execute-r-node")]
+        return generic_components
+
     @staticmethod
     def to_canvas_palette(components: List[Component]) -> dict:
         """
         Converts registry components into appropriate canvas palette format
         """
-        palette_json = list()
-
-        for component in components:
-            component_json = {
-                'label': component.name,
-                'image': "",
-                'id': component.id,
-                'description': component.description,
-                'node_types': [{
-                    'id': "",
-                    'op': component.id,
-                    'type': "execution_node",
-                    'inputs': [{
-                        "id": "inPort",
-                        "app_data": {
-                            "ui_data": {
-                                "cardinality": {'min': 0, 'max': -1},
-                                "label": "Input Port"
-                            }
-                        }
-                    }],
-                    'outputs': [{
-                        "id": "outPort",
-                        "app_data": {
-                            "ui_data": {
-                                "cardinality": {'min': 0, 'max': -1},
-                                "label": "Output Port"
-                            }
-                        }
-                    }],
-                    'parameters': {},
-                    'app_data': {
-                        'ui-data': {
-                            'label': component.name,
-                            'description': component.description,
-                            'image': "",
-                            'x_pos': 0,
-                            'y_pos': 0
-                        }
-                    }
-                }]
-            }
-
-            palette_json.append(component_json)
-        return palette_json
+        # Load jinja2 template
+        loader = PackageLoader('elyra', 'templates/components')
+        template_env = Environment(loader=loader)
+        template = template_env.get_template('canvas_palette_template.jinja2')
+
+        return template.render(components=components,
+                               num_components=len(components))
```

Comment:
Remove `num_components` here and use `components|length` in the template instead. Similar for `num_properties` in the properties palette function